### PR TITLE
Show path of expected sigdata in error message

### PR DESCRIPTION
**Pull request by forki**
https://github.com/Microsoft/visualfsharp/issues/1087 showed it's hard to debug sigdata issues.
This will at least give a small clue when the file is missing

### CODE REVIEW

**Comment by KevinRansom** [comment]
@forki   --- I like the idea of showing the directory.

Perhaps instead of just tweaking this we should fix it properly, I see four error messages when optdata is missing:
Each of them has the path at least once and often twice.

**With this PR:**
````
unknown(1,1): error FS1225: File 'd:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.optdata' not found alongside FSharp.Core. File expected in d:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.optdata.

unknown(1,1): error FS0229: Error opening binary file 'd:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.dll': File 'd:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.optdata' not found alongside FSharp.Core. File expected in d:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.optdata.

unknown(1,1): error FS3160: Problem reading assembly 'FSharp.Core, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a': Error opening binary file 'd:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.dll': File 'd:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.optdata' not found alongside FSharp.Core. File expected in d:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.optdata.

unknown(1,1): error FS0073: internal error: BuildFrameworkTcImports: no successful import of d:\microsoft\visualfsharp\release\net40\bin\FSharp.Core.dll
````

Originally (Awfull):
````
unknown(1,1): error FS1225: File 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.optdata' not found alongside FSharp.Core

unknown(1,1): error FS0229: Error opening binary file 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.dll': File 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.optdata' not found alongside FSharp.Core

unknown(1,1): error FS3160: Problem reading assembly 'FSharp.Core, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a': Error opening binary file 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.dll': File 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.optdata' not found alongside FSharp.Core

unknown(1,1): error FS0073: internal error: BuildFrameworkTcImports: no successful import of d:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.dll
```

Even the original sigdata shows the path if you know where to look 3 of the 4 error messages produced:
````
unknown(1,1): error FS1223: FSharp.Core.sigdata not found alongside FSharp.Core

unknown(1,1): error FS0229: Error opening binary file 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.dll': FSharp.Core.sigdata not found alongside FSharp.Core

unknown(1,1): error FS3160: Problem reading assembly 'FSharp.Core, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a': Error opening binary file 'd:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.dll': FSharp.Core.sigdata not found alongside FSharp.Core

unknown(1,1): error FS0073: internal error: BuildFrameworkTcImports: no successful import of d:\KevinRansom\visualfsharp\release\net40\bin\FSharp.Core.dll
````

**Comment by forki** [comment]
What do you want me to do? The original broken build didn't show the path right?

**Comment by KevinRansom** [comment]
Well ... the original emitted 4 different messages for sigdata, only the first didn't have the path, that is why we didn't notice it.  I think the right thing to do here, and it may be quite hard :-( , but one error should produce one message ... and that message should be sufficient to diagnose and fix the problem.  So if we could figure out how to reduce the error messages to one, which contains the path, at that point we should have a pretty decent thing.

a missing .optdata also produced 4 messages, all of which had the path.

Does that make sense ?

Kevin

File: src/fsharp/CompileOps.fs [hunk 1]
```diff
@@ -3565,9 +3565,9 @@ type RawFSharpAssemblyDataBackedByFileOnDisk (ilModule: ILModuleDef, ilAssemblyR
                 if List.contains ilShortAssemName externalSigAndOptData then 
                     let sigFileName = Path.ChangeExtension(filename, "sigdata")
                     if not sigDataReaders.IsEmpty then 
-                        error(Error(FSComp.SR.buildDidNotExpectSigdataResource(),m));
-                    if not (FileSystem.SafeExists sigFileName)  then 
-                        error(Error(FSComp.SR.buildExpectedSigdataFile(), m));
+                        error(Error(FSComp.SR.buildDidNotExpectSigdataResource(Path.GetFullPath filename),m));
```
Comment (dsyme) [inline src/fsharp/CompileOps.fs]: This should use ``FileSystem.GetFullPathShim``, likewise two lines below

Comment (forki) [inline src/fsharp/CompileOps.fs]: done

**Comment by dsyme** [comment]
@forki  Please merge with master and resolve conflicts?  Thanks

**Comment by forki** [comment]
done.

**Comment by dsyme** [comment]
Thanks, this is a really helpful contribution

**Pull request merged**